go/libkb/resolve.go

```diff
@@ -418,6 +425,62 @@ func (r *ResolverImpl) resolveTeamViaServerLookup(m MetaContext, au AssertionURL
 	return res
 }
 
+type phoneNumberLookup struct {
+	AppStatusEmbed
+	User *keybase1.PhoneLookupResult `json:"user"`
+}
+
+func (r *ResolverImpl) resolvePhoneNumberViaServerLookup(m MetaContext, au AssertionURL, input string) (res ResolveResult) {
+	m.CDebugf("resolvePhoneNumberViaServerLookup")
+
+	key, val, err := au.ToLookup()
+	if err != nil {
+		res.err = err
+		return res
+	}
+
+	if key != "phone" {
+		res.err = fmt.Errorf("expected 'phone' assertion")
+		return res
+	}
+
+	arg := NewAPIArgWithMetaContext(m, "user/phone_numbers_search")
+	arg.SessionType = APISessionTypeREQUIRED
+	arg.Args = make(HTTPArgs)
+	arg.Args["phone_number"] = S{Val: val}
+	arg.AppStatusCodes = []int{SCOk}
+
+	var lookup phoneNumberLookup
+	if err := m.G().API.GetDecode(arg, &lookup); err != nil {
+		if appErr, ok := err.(AppStatusError); ok {
+			switch appErr.Code {
+			case SCInputError:
+				err = ResolutionError{Input: input, Msg: err.Error(), Kind: ResolutionErrorInvalidInput}
+			// TODO: Phone number search rate limiting code goes here.
```

Comment:
@mlsteele this one is being done in another ticket, left a TODO so it's easier to point someone where to look about changing code in the client

Reply:
sounds good. Can be nice to have the ticket number number in the todo